### ISSUE TITLE
Wrong number of parameters for denote-link

### ISSUE THREAD

**Issue by qingshuizheng** [pprevos/citar-denote#25]
https://github.com/pprevos/citar-denote/blob/f093c37d28320a04b5e7ee87d5c442fefa749c35/citar-denote.el#L487

Hello @pprevos , 

Seems `denote-link` require at least 3 parameters here, but only 1 are offered here.

```
(denote-link FILE FILE-TYPE DESCRIPTION &optional ID-ONLY)
```

Zheng

**Comment by pprevos** [comment]
Hi Zheng,

I am working on a new version to fix compatability with the latest Denote versions.

This will be released in the next version, can you please test:

```
(defun citar-denote-link-reference ()
  "Insert a Denote link to a bibliographic note using Citar selection."
  (interactive)
  (if (denote-file-is-note-p (buffer-file-name))
      (let* ((citekey (citar-select-refs
		       :filter (citar-denote--has-notes)
		       :multiple nil))
	     (files (gethash (car citekey)
			     (citar-denote--get-notes citekey)))
	     (file (if (= (length files) 1)
		       (car files)
		     (citar-denote--select-file-using-title files)))
             (file-type (denote-filetype-heuristics file))
             (description (read-string
                           "Description: "
                           (denote--link-get-description file file-type))))
        (denote-link file file-type description))
    (user-error "Buffer is not a Denote file")))
```

**Comment by pprevos** [comment]
Hi Zheng, the latest release fixes this issue ([0c68a4f](https://github.com/pprevos/citar-denote/commit/0c68a4f74f5dcfc23ef03b211658a2ccc5d9aa3b)), plus a range of other improvements.

**Comment by qingshuizheng** [comment]
Thanks @pprevos . It works great. 👍
